lib/puppet/provider/foreman_resource/rest_v3.rb

```diff
@@ -95,6 +98,21 @@ def success?(response)
   end
 
   def error_message(response)
-    JSON.parse(response.body)['error']['full_messages'].join(' ') rescue "unknown error (response #{response.code})"
+    messages = {
+      '400' => '400 Bad Request: Something is wrong with the data we sent to Foreman server',
```

Comment:
Probably better to extract it a bit, but you can do:

```suggestion
      '400' => "#{response.code} #{response.message}: Something is wrong with the data we sent to Foreman server",
```

That way you can provide it for every error code even if there is no detailed explanation.

Reply:
Thanks. I updated the general case to include the #{response.message} as well. (I also removed the `unknown error` bit).`

Reply:
I don't consider this point resolved. What I meant is that on every single line you shouldn't repeat the code and response message. You can even simplify this (as I hinted at with 'extract it a bit').
```ruby
explanations = {
  '400' => 'Something is wrong with the data we sent to Foreman server',
  ...
}

if (explanation = explanations[response.code])
  "#{response.code} #{response.message}: #{explanation}"
else
  ...
end
```

Reply:
Thanks for the suggestion. I'm going to push an update once tests are passing for me that incorporates this approach